_posts/22-06-2021-kubevirt-ipv6-config-plus-ssh-key-injection.md

```diff
@@ -0,0 +1,119 @@
+---
+title: "KubeVirt: IPv6 configuration plus ssh public key injection"
+date: 2021-04-22
+categories:
+  - blog
+tags:
+  - ipv6
+  - cloud-init
+---
+
+In this short post, I will show how to configure IPv6 for a VM running in
+KubeVirt, and inject an ssh public key into it, via cloud-init.
+
+The first question I should answer, is "why?"; shouldn't the reader simply
+look at the
+[documentation](https://kubevirt.io/user-guide/virtual_machines/accessing_virtual_machines/#static-ssh-key-injection-via-cloud-init)
+for how to inject a public ssh key into the KubeVirt VM, extend that yaml
+specification with the network configuration found in the
+[KubeVirt examples](https://github.com/kubevirt/kubevirt/blob/ccfdfd80a12a18963e23cd131582134e3bd9b974/examples/vmi-masquerade.yaml#L40)
+and be done with it ?
+
+The short answer is: "that will not work". The reason is simple; cloud-init
+[only supports](https://cloudinit.readthedocs.io/en/latest/topics/network-config.html#network-configuration-sources)
+the netplan format for the `NoCloud` datasource. Furthermore, the `ConfigDrive`
+datasource implementation in KubeVirt is limited to storing a
+`network_data.json` file, which would leave us with a nasty looking Openstack
+metadata format looking network configuration.
+
+While that would work, it would lead to more questions than answers - yes; we
+would need to comply with the Openstack metadata network_data schema, which,
+quite simply, does not make sense outside the openstack world.
+
+As such, the simplest (I like simple) way out, is to abuse the `user-data`
```

Comment:
Is this an abuse?

Maybe you could word it as we are forced to write an OS specific script to configure the network instead of using abstract netplan.

Reply:
Done.